### ISSUE TITLE
Consecutive lines do not render contiguously; sometimes stutters

### ISSUE THREAD

**Issue by cubetastic33** [alacritty/alacritty#4647]
### System

OS: Arch Linux
Version: alacritty 0.6.0 (04cbf767)
Display server: Xorg
WM: bspwm
Compositor: picom

### Logs

Output of `alacritty -vv`:
```
[2021-01-02 18:35:24.318505071] [INFO ] [alacritty] Welcome to Alacritty
Created log file at "/tmp/Alacritty-68564.log"
[2021-01-02 18:35:24.318567778] [INFO ] [alacritty] Configuration files loaded from:
[2021-01-02 18:35:24.318577164] [INFO ] [alacritty]   "/home/aravk/.config/alacritty/alacritty.yml"
[2021-01-02 18:35:24.345990416] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[2021-01-02 18:35:24.348248646] [DEBUG] [alacritty] Estimated DPR: 1.0833333333333333
[2021-01-02 18:35:24.348278376] [DEBUG] [alacritty] Estimated window size: None
[2021-01-02 18:35:24.348293592] [DEBUG] [alacritty] Estimated cell size: 7 x 14
[2021-01-02 18:35:24.440889583] [INFO ] [alacritty] Device pixel ratio: 1.0833333333333333
[2021-01-02 18:35:24.448346349] [INFO ] [alacritty] Initializing glyph cache...
[2021-01-02 18:35:24.451699609] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[2021-01-02 18:35:24.460857859] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[2021-01-02 18:35:24.466506265] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[2021-01-02 18:35:24.475695330] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[2021-01-02 18:35:24.486031915] [INFO ] [alacritty] ... finished initializing glyph cache in 0.037641797s
[2021-01-02 18:35:24.486711519] [INFO ] [alacritty] Cell size: 7 x 14
[2021-01-02 18:35:24.486791093] [INFO ] [alacritty] Padding: 18 x 13
[2021-01-02 18:35:24.486807723] [INFO ] [alacritty] Width: 800, Height: 600
[2021-01-02 18:35:24.502662745] [INFO ] [alacritty] PTY dimensions: Line(41) x Column(109)
[2021-01-02 18:35:24.518941028] [INFO ] [alacritty] Initialisation complete
[2021-01-02 18:35:24.526645078] [DEBUG] [alacritty_terminal] New num_cols is 190 and num_lines is 50
[2021-01-02 18:35:24.531818753] [INFO ] [alacritty] Padding: 18 x 13
[2021-01-02 18:35:24.531913921] [INFO ] [alacritty] Width: 1366, Height: 727
[2021-01-02 18:35:24.609842315] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['1','3',],['#','a','b','b','2','b','f',],] at line 797
[2021-01-02 18:35:24.609874955] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['1','7',],['#','a','b','b','2','b','f',],] at line 797
[2021-01-02 18:35:24.609893640] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['1','9',],['#','2','5','2','5','3','2',],] at line 797
[2021-01-02 18:35:24.609907979] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['4',],['2','5','6',],['#','a','b','b','2','b','f',],] at line 797
[2021-01-02 18:35:24.609920563] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['7','0','8',],['#','2','5','2','5','3','2',],] at line 797
[2021-01-02 18:35:24.614302974] [DEBUG] [alacritty_terminal] [unhandled osc_dispatch]: [['1',],['~',],] at line 797
[2021-01-02 18:35:24.620843180] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[2021-01-02 18:35:44.892090469] [INFO ] [alacritty] Goodbye
Deleted log file at "/tmp/Alacritty-68564.log"
```

### Description

When using terminal visualizers, like cava or the one in ncmpcpp, consecutive lines have space between them even when the y offset is 0 in the config file. Besides this, the time display feature in ncmpcpp stutters on alacritty. Both of these issues aren't there on kitty. Here's a video demonstrating the issues; the top terminal is alacritty, and the bottom one is kitty.
 

https://user-images.githubusercontent.com/33519300/103458378-592e3680-4d2d-11eb-905e-955c7cd8a998.mp4

**Comment by chrisduerr** [comment]
Alacritty just uses your font and its metrics to draw these characters, instead of rendering them completely independently like Kitty. So yes, depending on your font that might cause some troubles.

The flickering is likely do to Alacritty rewriting the screen while the application is still updating it, it might not update it very efficiently and Alacritty just renders things as soon as it is instructed to do so. A fix for that is tracked in #598.

**Comment by cubetastic33** [comment]
Ah, I see. Thanks a lot.
